data/secscan_model/test/test_secscan_v4_model.py

```diff
@@ -1,26 +1,154 @@
+        repository=repository_ref._db_id,
+        error_json={},
+        index_status=IndexStatus.COMPLETED,
+        indexer_hash="abc",
+        indexer_version=IndexerVersion.V4,
+        metadata_json={},
+    )
+    mss.save()
+
+    secscan = V4SecurityScanner(app, instance_keys, storage)
+    secscan._secscan_api = mock.Mock()
+    secscan._secscan_api.vulnerability_report.return_value = None
+
+    assert secscan.load_security_information(manifest).status == ScanLookupStatus.NOT_YET_INDEXED
+
+
+def test_load_security_information_success(initialized_db):
+    repository_ref = registry_model.lookup_repository("devtable", "simple")
+    tag = registry_model.get_repo_tag(repository_ref, "latest", include_legacy_image=True)
+    manifest = registry_model.get_manifest_for_tag(tag, backfill_if_necessary=True)
+
+    mss = ManifestSecurityStatus(
+        manifest=manifest._db_id,
+        repository=repository_ref._db_id,
+        error_json={},
+        index_status=IndexStatus.COMPLETED,
+        indexer_hash="abc",
+        indexer_version=IndexerVersion.V4,
+        metadata_json={},
+    )
+    mss.save()
+
+    secscan = V4SecurityScanner(app, instance_keys, storage)
+    secscan._secscan_api = mock.Mock()
+    secscan._secscan_api.vulnerability_report.return_value = {
+        "manifest_hash": manifest.digest,
+        "state": "IndexFinished",
+        "packages": {},
+        "distributions": {},
+        "repository": {},
+        "environments": {},
+        "package_vulnerabilities": {},
+        "success": True,
+        "err": "",
+    }
+
+    result = secscan.load_security_information(manifest)
+
+    assert result.status == ScanLookupStatus.SUCCESS
+    assert result.security_information == {"Layer": {"Features": []}}
+
+
 def test_perform_indexing_whitelist(initialized_db):
     app.config["SECURITY_SCANNER_V4_NAMESPACE_WHITELIST"] = ["devtable"]
+    expected_manifests = (
+        Manifest.select().join(Repository).join(User).where(User.username == "devtable")
+    )
+
     secscan = V4SecurityScanner(app, instance_keys, storage)
+    secscan._secscan_api = mock.Mock()
+    secscan._secscan_api.state.return_value = "abc"
+    secscan._secscan_api.index.return_value = (
+        {"err": None, "state": IndexReportState.INDEX_FINISHED},
+        "abc",
+    )
 
     next_token = secscan.perform_indexing()
 
+    assert secscan._secscan_api.index.call_count == expected_manifests.count()
+    for mss in ManifestSecurityStatus.select():
+        assert mss.repository.namespace_user.username == "devtable"
+    assert ManifestSecurityStatus.select().count() == expected_manifests.count()
     assert (
         Manifest.get_by_id(next_token.min_id - 1).repository.namespace_user.username == "devtable"
     )
 
 
 def test_perform_indexing_no_whitelist(initialized_db):
+    app.config["SECURITY_SCANNER_V4_NAMESPACE_WHITELIST"] = []
     secscan = V4SecurityScanner(app, instance_keys, storage)
+    secscan._secscan_api = mock.Mock()
```

Comment:
Please no mocks... They are *incredibly* fragile. I'd much prefer a fake.

Reply:
We aren't using the mock extensively here, just for one return value to test a single code path. Writing an entire fake seems like overkill for this specific test scenario (I did write a fake for `test_v4_secscan.py` however).

Reply:
K